DataFormats/Common/interface/DetSetVectorNew.h

```diff
@@ -23,6 +23,8 @@ namespace edm { namespace refhelper { template<typename T> struct FindForNewDetS
 namespace edmNew {
   typedef uint32_t det_id_type;
 
+  struct CapacityExaustedException {};
```

Comment:
At the very least throw an exception that inherits from ```std::exception``` so that at the unlikely event the exception escapes we can still call ```.what()```.